public/app/features/alerting/unified/state/AlertmanagerContext.tsx

```diff
@@ -99,3 +98,15 @@ function isAlertManagerAvailable(availableAlertManagers: AlertManagerDataSource[
   const availableAlertManagersNames = availableAlertManagers.map((am) => am.name);
   return availableAlertManagersNames.includes(alertManagerName);
 }
+
+// when the `implementation` is set to `undefined` we assume we're dealing with an AlertManager with config API. The reason for this is because
+// our Hosted Grafana stacks provision Alertmanager data sources without `jsonData: { implementation: "mimir" }`.
+const CONFIG_API_ENABLED_ALERTMANAGER_FLAVORS = [
+  AlertManagerImplementation.mimir,
+  AlertManagerImplementation.cortex,
+  undefined,
+];
+
+export function isAlertManagerWithConfigAPI(dataSourceConfig: AlertManagerDataSourceJsonData): boolean {
+  return CONFIG_API_ENABLED_ALERTMANAGER_FLAVORS.includes(dataSourceConfig.implementation!);
```

Comment:
Is the exclamation mark for `implementation` necessary?

Reply:
Not any more, since `undefined` is a valid value now – good catch!